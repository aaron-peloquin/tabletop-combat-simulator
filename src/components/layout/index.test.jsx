// import { Provider } from "react-redux"
import {createShallow} from "@material-ui/core/test-utils"
import toJson from "enzyme-to-json"

import Layout from "./index"

describe("<Layout />", async ()=>{
  const renderShallowUntilComponent = createShallow({"untilSelector": "Layout"})
  let props; let Component
  beforeEach(async ()=>{
    Component = await renderShallowUntilComponent(<Layout {...props} />)
  })

  it("loads", () => {
    expect(typeof Component).toBe("object")
  })

  it("has a <HeaderBar />, <SideBar /> and <main />", () => {
    expect(Component.find("Connect(HeaderBar)").length).toBe(1)
    expect(Component.find("Connect(WithStyles(SideBar))").length).toBe(1)
    expect(Component.find("main").length).toBe(1)
  })

  it("snapshots", () => {
    expect(toJson(Component)).toMatchSnapshot()
  })
})
