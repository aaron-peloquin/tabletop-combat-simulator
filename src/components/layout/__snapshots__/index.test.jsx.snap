// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> snapshots 1`] = `
<div>
  <HeaderBar />
  <Connect(WithStyles(SideBar)) />
  <main
    className="Layout-content-1"
  >
    <WithStyles(Grid)
      container={true}
    >
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <WithStyles(Paper)
          className="Layout-paper-2"
        >
          No page content was passed
        </WithStyles(Paper)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </main>
</div>
`;
